src/shogun/mathematics/Statistics.cpp

```diff
@@ -325,6 +325,7 @@ SGVector<int32_t> CStatistics::sample_indices(int32_t sample_size, int32_t N)
 	int32_t* idxs=SG_MALLOC(int32_t,N);
 	int32_t i, rnd;
 	int32_t* permuted_idxs=SG_MALLOC(int32_t,sample_size);
+	auto rng = std::unique_ptr<CRandom>(new CRandom(sg_random_seed));
```

Comment:
@MikeLing do you agree that `CRandom(sg_random_seed)` and `CRandom()` will construct the same object with the same state? because it does...
so there's really no need for
`new CRandom(sg_random_seed))` it's enough to call `new CRandom()` right?